### PR TITLE
feat: add scheme field test case

### LINKED ISSUE #1642: Whether to consider adding scheme configuration items in the upstream configuration
# Feature request

## Please describe your feature

In the current configuration page related to `upstream`, there is no scheme configuration item yet, but there is a `scheme` field in the upstream object of APISIX. Do we need to increase the configuration of this field in the Dashboard?
For the scheme field, please refer to: https://github.com/apache/apisix/blob/master/docs/en/latest/admin-api.md#upstream

## Describe the solution you'd like

The configuration of the `scheme` field should be supported in the upstream configuration.

### PR DESCRIPTION
**Pull request by batman-ezio**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

add an optional field scheme on the upstream form

**Related issues**

fix/resolve #1642 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
cc @guoqqqi to take a look

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 1]
```diff
@@ -694,6 +694,25 @@ const UpstreamForm: React.FC<Props> = forwardRef(
                 })}
               </Select>
             </Form.Item>
+            <Form.Item
+              label={formatMessage({ id: 'page.upstream.step.scheme' })}
+              name="scheme"
+            >
+              <Select disabled={readonly}>
```
Comment (LiteSun) [inline web/src/components/Upstream/UpstreamForm.tsx:701]:
```suggestion
              <Select disabled={readonly}>
              {['http', 'https', 'grppc', 'grpcs'].map( item => (  
                  <Select.Option value={item}>
                     {item}
                   </Select.Option>))
              </Select>
```

Comment (juzhiyuan) [reply web/src/components/Upstream/UpstreamForm.tsx:701]: use `index` in `loop`

Comment (batman-ezio) [reply web/src/components/Upstream/UpstreamForm.tsx:701]: you mean replace `item` to `index`?

Comment (juzhiyuan) [reply web/src/components/Upstream/UpstreamForm.tsx:701]: image

here :)

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 2]
```diff
@@ -699,6 +699,11 @@ const UpstreamForm: React.FC<Props> = forwardRef(
               name="scheme"
             >
               <Select disabled={readonly}>
+              {['http', 'https', 'grppc', 'grpcs'].map( item => (  
+                  <Select.Option value={item}>
```
Comment (juzhiyuan) [inline web/src/components/Upstream/UpstreamForm.tsx]:
```suggestion
                  <Select.Option value={item} key={item}>
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 3]
```diff
@@ -694,6 +694,19 @@ const UpstreamForm: React.FC<Props> = forwardRef(
                 })}
               </Select>
             </Form.Item>
+            <Form.Item
+              label={formatMessage({ id: 'page.upstream.step.scheme' })}
+              name="scheme"
+            >
+              <Select disabled={readonly}>
+              {['http', 'https', 'grppc', 'grpcs'].map((item, index) => (
```
Comment (imjoey) [inline web/src/components/Upstream/UpstreamForm.tsx:702]: Hi @batman-ezio , as refer to https://github.com/apache/apisix/blob/master/docs/en/latest/admin-api.md#upstream, I guess the third one `grppc` is a typo here.

Comment (batman-ezio) [reply web/src/components/Upstream/UpstreamForm.tsx]: thanks. updated

**Comment by juzhiyuan** [comment]
also cc @guoqqqi to review.

**Review by guoqqqi — APPROVED** [review]
LGTM!

**Comment by membphis** [comment]
ci failed, please take a look: https://github.com/apache/apisix-dashboard/pull/1661/checks?check_run_id=2301814384#step:10:798

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 4]
```diff
@@ -688,6 +688,19 @@ const UpstreamForm: React.FC<Props> = forwardRef(
                 })}
               </Select>
             </Form.Item>
+            <Form.Item
+              label={formatMessage({ id: 'page.upstream.step.scheme' })}
+              name="scheme"
+            >
+              <Select disabled={readonly}>
+              {['http', 'https', 'grpc', 'grpcs'].map((item, index) => (
```
Comment (jbampton) [inline web/src/components/Upstream/UpstreamForm.tsx:696]: I have never done `.tsx` before.  I have done one other `.tsx` review.  So similar my view is that an array or list should be ordered somehow and mostly its done alphabetically.  I always like rules, standards and style guides. 

```suggestion
              {['grpc', 'grpcs', 'http', 'https'].map((item, index) => (
```

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 5]
```diff
@@ -688,6 +688,19 @@ const UpstreamForm: React.FC<Props> = forwardRef(
...
+                <Select.Option value={item} key={index}>
```
Comment (jbampton) [inline web/src/components/Upstream/UpstreamForm.tsx:697]: In computer science, an associative array, map, symbol table, or dictionary is an abstract data type composed of a collection of (key, value) pairs, such that each possible key appears at most once in the collection

https://en.wikipedia.org/wiki/Associative_array

So my view is that `key` should come before `value` and also based on order alphabetically.

```suggestion
                <Select.Option key={index} value={item}>
```

Comment (juzhiyuan) [reply web/src/components/Upstream/UpstreamForm.tsx:697]: That's fine, and not urgent 👍

**Comment by juzhiyuan** [comment]
I know what happened, one PR needs to be fixed.

**Comment by juzhiyuan** [comment]
just wait for fixing it 😂

**Comment by nic-chen** [comment]
@batman-ezio  maybe you could sync code from master to try to fix FE CI

**Pull request closed without merging**